### PR TITLE
Documentation/paul

### PR DESCRIPTION
**Pull request by pmcintyr**

### CODE REVIEW

File: app/src/main/java/com/github/brugapp/brug/ui/QrCodeScannerActivity.kt [hunk 1]
```diff
@@ -78,13 +83,19 @@ class QrCodeScannerActivity : AppCompatActivity() {
...
+    /**
+     * TODO
+     *
+     */
     override fun onResume() {
         super.onResume()
         viewModel.startPreview()
     }
 
+    /**
```
Comment (DeepGreen1) [inline app/src/main/java/com/github/brugapp/brug/ui/QrCodeScannerActivity.kt:95]: Why did you add those "TODO" comments? (same question for onResume function)

File: app/src/main/java/com/github/brugapp/brug/ui/NFCScannerActivity.kt [hunk 2]
```diff
@@ -111,6 +111,12 @@ open class NFCScannerActivity: AppCompatActivity() {
         }
     }
 
+    /**
+     * TODO
```
Comment (DeepGreen1) [inline app/src/main/java/com/github/brugapp/brug/ui/NFCScannerActivity.kt:115]: This file has a lot of "TODOS". Could you be more specific on what the method do?

File: app/src/main/res/layout/activity_sign_in.xml [hunk 3]
```diff
@@ -1,98 +1,189 @@
...
+
+     <ProgressBar 
+
+             android:id="@+id/loadingUser" 
+
+             android:layout_width="wrap_content" 
+
+             android:layout_height="wrap_content" 
+
+             app:layout_constraintTop_toTopOf="parent"
```
Comment (DeepGreen1) [inline app/src/main/res/layout/activity_sign_in.xml:173]: I think that the code is identical here but with lots more of spaces. Did the AndroidStudio tool do that?

**Review by DeepGreen1 — APPROVED** [review]
Nice work on the documentation. I'll let you remove the "TODO" and if possible remove the blank spaces in the xml (check the comments). Otherwise everything seems fine! You can merge when you're ready.

**Review by Mouniraki — CHANGES_REQUESTED** [review]
You did a decent job, but you have many comments which should be re-written or completely deleted. For instance, every private function should not be commented, and every comment should be complete, in the sense where in addition to the concise explanation of what your function does, every argument should be described, as well as the return value if there is one.

File: app/src/main/java/com/github/brugapp/brug/ui/QrCodeScannerActivity.kt (hunk 1 above)
Comment (pmcintyr) [reply app/src/main/java/com/github/brugapp/brug/ui/QrCodeScannerActivity.kt:92]: I have removed them in my latest commit

File: app/src/main/java/com/github/brugapp/brug/ui/NFCScannerActivity.kt (hunk 2 above)
Comment (pmcintyr) [reply app/src/main/java/com/github/brugapp/brug/ui/NFCScannerActivity.kt]: Yes I have updated the comments correspondingly in my latest commit

File: app/src/main/res/layout/activity_sign_in.xml (hunk 3 above)
Comment (pmcintyr) [reply app/src/main/res/layout/activity_sign_in.xml]: good catch, I have removed the blank lines!

File: app/src/main/java/com/github/brugapp/brug/ui/AddItemActivity.kt [hunk 4]
```diff
@@ -101,6 +118,15 @@ class AddItemActivity : AppCompatActivity() {
...
 
+    /**
+     * we create and add item to current user's inventory
+     *
+     * @param itemName
+     * @param itemNameHelper
+     * @param itemType
+     * @param itemDesc
+     * @param firebaseAuth
+     */
```
[the remark this answers is not in the document]
Comment (pmcintyr) [reply app/src/main/java/com/github/brugapp/brug/ui/AddItemActivity.kt]: We write documentation for potential future developers that need to undestand our code, I would also like them to understand my private functions. I am not explaining the functionality of a public API but the functionality of core methods necessary to our features. [Here is a link to why documenting private functions is still a good practice.](https://stackoverflow.com/questions/1743538/should-i-document-my-private-methods#:~:text=Private%20functions%20are%20especially%20important,developer%20can%20infer%20their%20invariants.)

File: app/src/main/java/com/github/brugapp/brug/ui/NFCScannerActivity.kt [hunk 5]
```diff
@@ -111,6 +111,12 @@ open class NFCScannerActivity: AppCompatActivity() {
         }
     }
 
+    /**
+     * TODO
+     *
+     * @param editable
+     * @return
+     */
```
[the remark this answers is not in the document]
Comment (pmcintyr) [reply app/src/main/java/com/github/brugapp/brug/ui/NFCScannerActivity.kt]: Good idea, this is now removed

File: app/src/main/java/com/github/brugapp/brug/view_model/NFCScanViewModel.kt [hunk 6]
```diff
@@ -27,23 +27,27 @@ private const val NFC_REQUEST_CODE = 1000101
 
 class NFCScanViewModel : ViewModel() {
     /**
+     * makes sure we have NFC permissions
      * @param context1
      */
```
[the remark this answers is not in the document]
Comment (pmcintyr) [reply app/src/main/java/com/github/brugapp/brug/view_model/NFCScanViewModel.kt]: nice catch, i had a "this1" parameter that couldn't be renamed to "this" for obvious reasons but i have renamed these parameters with unnecessary concatenated numbers

File: app/src/main/java/com/github/brugapp/brug/view_model/NFCScanViewModel.kt [hunk 7]
```diff
@@ -27,23 +27,27 @@ private const val NFC_REQUEST_CODE = 1000101
...
+     * makes sure we have NFC permissions
      * @param context1
      */
     fun checkNFCPermission(context1: Context){ if(PackageManager.PERMISSION_DENIED == checkSelfPermission(context1,Manifest.permission.NFC)) requestPermissions(context1 as Activity, arrayOf(Manifest.permission.NFC), NFC_REQUEST_CODE) }
 
     /**
+     * initializes Nfc adapter if it exists
      * @param this1
      * @return
      */
```
[the remark this answers is not in the document]
Comment (pmcintyr) [reply app/src/main/java/com/github/brugapp/brug/view_model/NFCScanViewModel.kt]: this has now been corrected

File: app/src/main/res/layout/activity_sign_in.xml [hunk 8]
```diff
@@ -1,98 +1,189 @@
...
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintLeft_toLeftOf="parent"
-            app:layout_constraintRight_toRightOf="parent"
-            android:indeterminate="true"
-            android:visibility="invisible"/>
-
-</androidx.constraintlayout.widget.ConstraintLayout>
+<?xml version="1.0" encoding="utf-8"?> 
 
+ <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (pmcintyr) [reply app/src/main/res/layout/activity_sign_in.xml]: this has now been corrected

File: app/src/main/java/com/github/brugapp/brug/ui/QrCodeScannerActivity.kt [hunk 9]
```diff
@@ -34,6 +34,11 @@ class QrCodeScannerActivity : AppCompatActivity() {
     @Inject
     lateinit var firebaseAuth: FirebaseAuth
 
+    /**
+     * TODO
+     *
+     * @param savedInstanceState
+     */
```
[the remark this answers is not in the document]
Comment (pmcintyr) [reply app/src/main/java/com/github/brugapp/brug/ui/QrCodeScannerActivity.kt]: this has been corrected also

File: app/src/main/java/com/github/brugapp/brug/ui/QrCodeScannerActivity.kt [hunk 10]
```diff
@@ -78,13 +81,17 @@ class QrCodeScannerActivity : AppCompatActivity() {
     }
 
 
-
-
+    /**
+     * we resume the Qr scan activity
+     */
```
Comment (Mouniraki) [inline app/src/main/java/com/github/brugapp/brug/ui/QrCodeScannerActivity.kt:86]: These comments are not necessary, since they override Android API functions which are already commented. You should remove them if they don't add anything more than what the Android API comments already explain.

**Review by Mouniraki — APPROVED** [review]
Consider my comments before merging your part to the main branch.

File: app/src/main/java/com/github/brugapp/brug/view_model/NFCScanViewModel.kt [hunk 11]
```diff
@@ -72,8 +78,9 @@ class NFCScanViewModel : ViewModel() {
     }
 
     /**
+     * builds the array of Ndefmessages from the Nfc adapter
      * @use sets up the data and conditions needed for calling buildTagViews
-     * @param intent used for getting extended data from this intent
+     * @param intent: used for getting extended data from this intent
      * @return Boolean: this object checks if the extended data from intent is null
      * @return Array<NdefMessage>: this object represented the extended data from intent
```
[the remark this answers is not in the document]
Comment (pmcintyr) [reply app/src/main/java/com/github/brugapp/brug/view_model/NFCScanViewModel.kt:85]: It returns a Pair of 2 elements: so both a boolean and an array

File: app/src/main/java/com/github/brugapp/brug/ui/QrCodeScannerActivity.kt (hunk 10 above)
Comment (pmcintyr) [reply app/src/main/java/com/github/brugapp/brug/ui/QrCodeScannerActivity.kt]: i agree, this is now removed
